Signal/Attachments/AttachmentDownloadRetryRunner.swift

```diff
@@ -81,10 +81,10 @@ public class AttachmentDownloadRetryRunner {
         }
 
         private func run() async {
-            do {
-                self.isRunning = true
-                defer { self.isRunning = false }
-
+            self.isRunning = true
+            defer { self.isRunning = false }
+            // Run while there is a next timestamp.
+            while true {
```

Comment:
This loop never breaks?

Reply:
It can break. If there is no more timestamps to handle it returns on line 92. 
Functionally it is the same as before.